web/components/chat/Chat.tsx

```diff
@@ -0,0 +1,114 @@
+'use client';
+import { useEffect, useMemo, useRef, useState } from 'react';
+import Message from './Message';
+import { useChatContext } from './ChatContext';
+import { ChatProps, DBMessage } from './types';
+import { ChatSkeleton } from './Skeletons';
+
+export default function Chat({
+  conversationId,
+  oldMessages,
+  currentUser,
+}: ChatProps) {
+  const [messages, setMessages] = useState<DBMessage[]>(
+    oldMessages as DBMessage[]
+  );
+  const { newMessages } = useChatContext();
+  const scrollContainerRef = useRef<HTMLDivElement>(null);
+  const [loading, setLoading] = useState(true);
+
+  const memoMessages = useMemo(() => {
+    // Sort the messages by date & time (oldest first)
+    return messages?.sort((a, b) => {
+      return a.created_at > b.created_at ? 1 : -1;
+    });
+  }, [messages]);
+
+  function addMessages(messages: DBMessage[]) {
+    // set messages with new messages but avoid duplicates
+    setMessages((prevMessages) => [
+      ...(prevMessages ?? []),
+      ...messages.filter((msg) => !prevMessages?.find((m) => m.id === msg.id)),
+    ]);
+  }
+
+  useEffect(() => {
+    // initialize messages with old messages
+    if (oldMessages) {
+      addMessages(oldMessages);
+    }
+  }, [oldMessages]);
+
+  useEffect(() => {
+    // add new messages to the list when fetched
+    if (newMessages && conversationId) {
+      const conversationNewMessages = newMessages[conversationId];
+
+      if (conversationNewMessages) {
+        if (conversationNewMessages.length > 0) {
+          addMessages(conversationNewMessages);
+        }
+        // stop loading when new messages are fetched
+        setLoading(false);
+      }
+    }
+  }, [newMessages, conversationId]);
+
+  useEffect(() => {
+    // Scroll to the bottom of the chat when loading is done
+    if (scrollContainerRef.current) {
+      scrollContainerRef.current.scrollTop =
+        scrollContainerRef.current.scrollHeight;
+    }
+  }, [loading]);
+
+  // Check if there is a conversation selected
+  if (!conversationId) {
+    return (
+      <div className='flex h-full w-full items-center justify-center dark:text-white'>
+        Select a conversation
+      </div>
+    );
+  }
```

Comment:
Ici tu pourrais call le notFound() de nextjs et creer une page not-found.tsx pour afficher le Conversation not found please select a valid one in the list